### PR TITLE
Ensure `index.mapping.single_type` can only be set on 5.x indices

### PR DESCRIPTION
**Pull request by s1monw**
In 6.x we prevent multiple types and default to `index.mapping.single_type: false`
This change removes the registered setting and ensures that it's preserved for
5.x indices.

Relates to #24961

Note this might still fail since not all places are cut over

### CODE REVIEW

File: qa/rolling-upgrade/src/test/resources/rest-api-spec/test/upgraded_cluster/10_basic.yml [hunk 1]
```diff
@@ -17,6 +17,19 @@
 
  - match: { hits.total: 5 } # just check we recovered fine
 
+
+ - do:
```
Comment (nik9000) [inline qa/rolling-upgrade/src/test/resources/rest-api-spec/test/upgraded_cluster/10_basic.yml]: :+1:

**Review by nik9000 — APPROVED** [review]
LGTM anytime the tests are ready for it.

**Review by jpountz — APPROVED** [review]
LGTM!

**Comment by jamshid** [comment]
Sorry I know this is closed but in case someone else ends up here from a web search...

Seems https://www.elastic.co/guide/en/elasticsearch/reference/6.7/removal-of-types.html falsely  claims that `index.mapping.single_type: true` continues to work with 6.x. Not only is that the default value, but the PUT request is not allowed to specify this value.

> The following examples can be used in Elasticsearch 5.6 or Elasticsearch 6.x. In 6.x, there is no need to specify index.mapping.single_type as that is the default.

This *fails* against ES 6.8.3 with a 400 error:
```
curl -X PUT -H 'Content-type: application/json' http://ELASTICSEARCH:9200/index1 --data-binary '
{
  "settings": {
    "index.mapping.single_type": true
  },
  "mappings": {
    "_doc": {
      "properties": {
        "name": {
          "type": "text"
        },
        "user_name": {
          "type": "keyword"
        },
        "email": {
          "type": "keyword"
        }
      }
    }
  }
}
'

HTTP/1.1 400 Bad Request
Warning: 299 Elasticsearch-6.8.3-0c48c0e "[types removal] The parameter include_type_name should be explicitly specified in create index requests to prepare for 7.0. In 7.0 include_type_name will default to 'false', and requests are expected to omit the type name in mapping definitions."
content-type: application/json; charset=UTF-8
content-length: X
{
  "error": {
    "root_cause": [
      {
        "type": "remote_transport_exception",
        "reason": "[...][indices:admin/create]"
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "unknown setting [index.mapping.single_type] please check that any required plugins are installed, or check the breaking changes documentation for removed settings"
  },
  "status": 400
}
```

I tried adding `?include_type_name=true` to the url, the docs indicate that's needed for some other backward compatibility, but it does *not* help.

PS: having to add a parameter for a release, then remove it for the next release, makes it hard for an app to support multiple ES versions.

**Pull request merged**